src/App.js

```diff
@@ -28,8 +28,13 @@ import { CLIENT_DEFAULTS } from './utils/CLIENT_DEFAULTS';
 // LOCALIZATION
 import { getTextForLanguage } from './utils/getTextForLanguage';
 
+/** App component; main top-level component of the app */
 // Change HashRouter tags below to Router tags to turn off hash routing; only used to be compatible with GitHub Pages
 class App extends Component {
+  /**
+   * Create App component instance
```

Comment:
(Current discussion pending - is this description of React behavior a useful one?)

Reply:
Slight change: The first notes should actually go in the actual App description, not the constructor description. The constructor should be about anything special happening with the, obviously, the construction of the component. E.g. What confusing `state` properties it creates and why. And why did they have to be made to be so confusing, I guess.

Reply:
Btw, I just meant 'confusing state properties' in general. I'm not sure if there are or aren't any here. I'll have to check it out.

Reply:
In my opinion it's probably worth giving at least a short description for each state property (after all, there are only four), but at least `clients` and `devProps`. Also, maybe these descriptions could be directly before `this.state` is defined, rather than before the constructor?

Reply:
Interesting! How would jsdoc interpret that? I mean, where would it appear visually in the docs?

Reply:
@ethanbb

Reply:
I'm not sure; I was looking into whether there's a specific tag to use for class fields defined in the constructor last night, but I didn't find anything. Hopefully if the jsdoc library recognizes ES6 classes, it would know to associate `this.x` definitions within a class with that class - but we can test to make sure that works.